pkg/jsonmessage/jsonmessage.go

```diff
@@ -12,15 +12,98 @@ import (
 	"github.com/docker/docker/pkg/units"
 )
 
+type messageEntry struct {
+	Code int    // HTTP response code
+	Str  string // Text of message with Sprintf sub vars
+}
+
+type MessageMap map[string]messageEntry
+
+var msgs *MessageMap
+
 type JSONError struct {
-	Code    int    `json:"code,omitempty"`
-	Message string `json:"message,omitempty"`
+	error
+	MessageID string        `json:"id"`             // Unique ID per msg
+	Args      []interface{} `json:"args,omitempty"` // Substitution values
+	Code      int           `json:"-"`              // HTTP code (default 500)
+	Message   string        `json:"message,omitempty"`
+
+	// Message should not be used - its there for old code.  Instead the
+	// message text shoudl be generated dynamically based on the MesasgeID
+	// and the Args.
+}
+
+func SetMessageMap(daMap *MessageMap) {
+	msgs = daMap
 }
 
 func (e *JSONError) Error() string {
+	if e.MessageID != "" {
+		return T(e.MessageID, e.Args...)
+	}
 	return e.Message
 }
 
+/*
+func (e *JSONError) Code() int {
+	entry, ok := msgs[e.messageID]
+	if !ok {
+		entry = msgs["Generic"]
+	}
+
+	return entry.Code
+}
+*/
```

Comment:
Dead code

(not really reviewing, just browsing through this PR, so feel free to ignore :smile: )